pkg/ottl/ottlfuncs/func_flatten.go

```diff
@@ -47,52 +49,79 @@ func flatten[K any](target ottl.PMapGetter[K], p ottl.Optional[string], d ottl.O
 		prefix = p.Get()
 	}
 
+	conflict := false
+	if !c.IsEmpty() {
+		conflict = c.Get()
+	}
+
 	return func(ctx context.Context, tCtx K) (any, error) {
 		m, err := target.Get(ctx, tCtx)
 		if err != nil {
 			return nil, err
 		}
 
 		result := pcommon.NewMap()
-		flattenMap(m, result, prefix, 0, depth)
+		existingKeys := map[string]int{}
```

Comment:
Could we put these two and the variable to track conflicts (`conflict`/`resolveConflicts`) on a struct then put the `flatten*` methods on the struct? That way instead of passing each of these to each function call we can just implicitly pass them through the struct, and the overall argument list would be a bit shorter.